src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/Server/MetadataUtilsSmi.cs

```diff
@@ -526,6 +520,9 @@ internal static bool IsCompatible(SmiMetaData firstMd, SqlMetaData secondMd)
                     firstMd.Scale == secondMd.Scale &&
                     firstMd.CompareOptions == secondMd.CompareOptions &&
                     firstMd.LocaleId == secondMd.LocaleId &&
+#if NETFRAMEWORK
+                    firstMd.Type == secondMd.Type &&
+#endif
```

Comment:
Maybe it's been missed in the transition process! Could you verify if we could keep it the same in the both projects , please?
```suggestion
                    firstMd.Type == secondMd.Type &&
```